### DISCUSSION TITLE
Sort by cluster the same for everyone despite refining by hand?

### DISCUSSION THREAD

**Discussion by Rollbacke** [matiasdelellis/facerecognition#738] [Q&A] [answered]
Before spending time to install Face Recognition. I would like to ask few questions about how it works, which I was unable to find on the Wiki.
Currently, I was using Recognize, but not reliable enough on face detection (I spend a lot of time refining the clusters, and I don't even know if it's useful later). Then, I realized that sort by cluster is by user, which means that everything I do is only available to my user... Obviously, for this point, there is no solution, since it is "by design ".
So: if I use Face Recognition, are the clusters the same for everyone? If I spend time creating the clusters, refining them, sorting them, will this be visible to all my users?
Also, does sorting people misassigned into clusters improve detection accuracy subsequently?
Thanks !

**Comment by matiasdelellis** [comment, marked as answer]
> I realized that sort by cluster is by user, which means that everything I do is only available to my user... Obviously, for this point, there is no solution, since it is "by design ".

Well, We can say that it is a design decision. The logic of not sharing the groups is given by the fact that in the end the names are independent of each user.
Assuming you share all photos with your family, in your clusters you probably assign yourself by your first name. But on your siblings' account, they probably call you by some affectionate nickname, and your parents in another way.

This is the design decision, however, there is pending work to analyze the images shared only once.. which can be a problem.

> So: if I use Face Recognition, are the clusters the same for everyone? If I spend time creating the clusters, refining them, sorting them, will this be visible to all my users?

No. We are not at this point yet. However, it is unlikely to improve as you expect. The idea is to improve the clustering algorithm to make it easier,  but we do not aim to share the results.

> Also, does sorting people misassigned into clusters improve detection accuracy subsequently?

Fixing the faces by assigning them to the correct person is only responsible for separating them from the wrong cluster, and ensuring that they cannot be grouped incorrectly later, but is not taken into account for new clusters. Literally doing "This person is not such", informs to database that this face is not valid to regroup, but allows you to assign a name again because you still want to see this person's photo.

Thanks !

**Reply by Rollbacke** [reply]
Hi @matiasdelellis ,
Thank you for your frank and detailed response, which allowed me to understand better how Face Recognition (and Recognize) works.
So I'm going to wait for a solution to appear (but I know I shouldn't hope too much :D).
